File: scripts/seed-all.py
#!/usr/bin/env python3
"""Seed all 72 matches × 5 users with auto-retry on rate limits."""
import json, time, urllib.request, sys, os, random

API = os.getenv("API_URL", "http://localhost:8787")

USERS = [
    {"name": "F5",    "bias": "HOME", "conf": (65, 90)},
    {"name": "Aleko", "bias": "AWAY", "conf": (30, 60)},
    {"name": "Tita",  "bias": "DRAW", "conf": (50, 75)},
    {"name": "Rex",   "bias": "HOME", "conf": (80, 95)},
    {"name": "Nina",  "bias": "none", "conf": (40, 70)},
]

MATCHES = [
    ("2026-06-11","19:00","Mexico","South Africa","MEX-SOU"),
    ("2026-06-12","02:00","South Korea","Czechia","SOU-CZE"),
    ("2026-06-12","19:00","Canada","Bosnia-Herzegovina","CAN-BOS"),
    ("2026-06-13","01:00","United States","Paraguay","UNI-PAR"),
    ("2026-06-13","19:00","Qatar","Switzerland","QAT-SWI"),
    ("2026-06-13","22:00","Brazil","Morocco","BRA-MOR"),
    ("2026-06-14","01:00","Haiti","Scotland","HAI-SCO"),
    ("2026-06-14","04:00","Australia","Türkiye","AUS-TUR"),
    ("2026-06-14","17:00","Germany","Curaçao","GER-CUR"),
    ("2026-06-14","20:00","Netherlands","Japan","NET-JAP"),
    ("2026-06-14","23:00","Ivory Coast","Ecuador","IVO-ECU"),
    ("2026-06-15","02:00","Sweden","Tunisia","SWE-TUN"),
    ("2026-06-15","16:00","Spain","Cape Verde","SPA-CAP"),
    ("2026-06-15","19:00","Belgium","Egypt","BEL-EGY"),
    ("2026-06-15","22:00","Saudi Arabia","Uruguay","SAU-URU"),
    ("2026-06-16","01:00","Iran","New Zealand","IRA-NEW"),
    ("2026-06-16","19:00","France","Senegal","FRA-SEN"),
    ("2026-06-16","22:00","Iraq","Norway","IRA-NOR"),
    ("2026-06-17","01:00","Argentina","Algeria","ARG-ALG"),
    ("2026-06-17","04:00","Austria","Jordan","AUS-JOR"),
    ("2026-06-17","17:00","Portugal","Congo DR","POR-CON"),
    ("2026-06-17","20:00","England","Croatia","ENG-CRO"),
    ("2026-06-17","23:00","Ghana","Panama","GHA-PAN"),
    ("2026-06-18","02:00","Uzbekistan","Colombia","UZB-COL"),
    ("2026-06-18","16:00","Czechia","South Africa","CZE-SOU"),
    ("2026-06-18","19:00","Switzerland","Bosnia-Herzegovina","SWI-BOS"),
    ("2026-06-18","22:00","Canada","Qatar","CAN-QAT"),
    ("2026-06-19","01:00","Mexico","South Korea","MEX-SOU1"),
    ("2026-06-19","19:00","United States","Australia","UNI-AUS"),
    ("2026-06-19","22:00","Scotland","Morocco","SCO-MOR"),
    ("2026-06-20","00:30","Brazil","Haiti","BRA-HAI"),
    ("2026-06-20","03:00","Türkiye","Paraguay","TUR-PAR"),
    ("2026-06-20","17:00","Netherlands","Sweden","NET-SWE"),
    ("2026-06-20","20:00","Germany","Ivory Coast","GER-IVO"),
    ("2026-06-21","00:00","Ecuador","Curaçao","ECU-CUR"),
    ("2026-06-21","04:00","Tunisia","Japan","TUN-JAP"),
    ("2026-06-21","16:00","Spain","Saudi Arabia","SPA-SAU"),
    ("2026-06-21","19:00","Belgium","Iran","BEL-IRA"),
    ("2026-06-21","22:00","Uruguay","Cape Verde","URU-CAP"),
    ("2026-06-22","01:00","New Zealand","Egypt","NEW-EGY"),
    ("2026-06-22","17:00","Argentina","Austria","ARG-AUS"),
    ("2026-06-22","21:00","France","Iraq","FRA-IRA"),
    ("2026-06-23","00:00","Norway","Senegal","NOR-SEN"),
    ("2026-06-23","03:00","Jordan","Algeria","JOR-ALG"),
    ("2026-06-23","17:00","Portugal","Uzbekistan","POR-UZB"),
    ("2026-06-23","20:00","England","Ghana","ENG-GHA"),
    ("2026-06-23","23:00","Panama","Croatia","PAN-CRO"),
    ("2026-06-24","02:00","Colombia","Congo DR","COL-CON"),
    ("2026-06-24","19:00","Bosnia-Herzegovina","Qatar","BOS-QAT"),
    ("2026-06-24","19:00","Switzerland","Canada","SWI-CAN"),
    ("2026-06-24","22:00","Morocco","Haiti","MOR-HAI"),
    ("2026-06-24","22:00","Scotland","Brazil","SCO-BRA"),
    ("2026-06-25","01:00","Czechia","Mexico","CZE-MEX"),
    ("2026-06-25","01:00","South Africa","South Korea","SOU-SOU"),
    ("2026-06-25","20:00","Curaçao","Ivory Coast","CUR-IVO"),
    ("2026-06-25","20:00","Ecuador","Germany","ECU-GER"),
    ("2026-06-25","23:00","Japan","Sweden","JAP-SWE"),
    ("2026-06-25","23:00","Tunisia","Netherlands","TUN-NET"),
    ("2026-06-26","02:00","Paraguay","Australia","PAR-AUS"),
    ("2026-06-26","02:00","Türkiye","United States","TUR-UNI"),
    ("2026-06-26","19:00","Norway","France","NOR-FRA"),
    ("2026-06-26","19:00","Senegal","Iraq","SEN-IRA"),
    ("2026-06-27","00:00","Cape Verde","Saudi Arabia","CAP-SAU"),
    ("2026-06-27","00:00","Uruguay","Spain","URU-SPA"),
    ("2026-06-27","03:00","Egypt","Iran","EGY-IRA"),
    ("2026-06-27","03:00","New Zealand","Belgium","NEW-BEL"),
    ("2026-06-27","21:00","Croatia","Ghana","CRO-GHA"),
    ("2026-06-27","21:00","Panama","England","PAN-ENG"),
    ("2026-06-27","23:30","Colombia","Portugal","COL-POR"),
    ("2026-06-27","23:30","Congo DR","Uzbekistan","CON-UZB"),
    ("2026-06-28","02:00","Algeria","Austria","ALG-AUS"),
    ("2026-06-28","02:00","Jordan","Argentina","JOR-ARG"),
]

def pick_winner(bias):
    if bias == "HOME": return "HOME"
    if bias == "AWAY": return "AWAY"
    if bias == "DRAW": return "DRAW"
    return random.choice(["HOME", "AWAY", "DRAW"])

def predict(user, match, pick_val, confidence):
    date, time_, home, away, match_id = match
    body = json.dumps({
        "user": user, "matchId": match_id, "home": home, "away": away,
        "pick": pick_val, "confidence": confidence, "take": ""
    }).encode()
    req = urllib.request.Request(
        f"{API}/api/predict",
        data=body,
        headers={"Content-Type": "application/json"},
        method="POST"
    )
    try:
        with urllib.request.urlopen(req, timeout=30) as resp:
            return json.loads(resp.read())
    except urllib.error.HTTPError as e:
        return json.loads(e.read())
    except Exception as e:
        return {"error": str(e)}

def p(msg):
    print(msg, flush=True)

def main():
    total = len(MATCHES) * len(USERS)
    p(f"\n=== Seeding {len(MATCHES)} matches x {len(USERS)} users = {total} predictions ===")
    p(f"Delay: 8s between requests, auto-retry on 429\n")
    
    done = 0
    failed = 0
    
    for user in USERS:
        p(f"\n-- {user['name']} ({user['bias']} bias) --")
        for match in MATCHES:
            pw = pick_winner(user["bias"])
            conf = random.randint(user["conf"][0], user["conf"][1])
            
            for attempt in range(5):
                result = predict(user["name"], match, pw, conf)
                err = result.get("error", "")
                if not err:
                    done += 1
                    p(f"  OK {user['name']}: {match[4]} -> {pw} ({conf}%) [{done}/{total}]")
                    break
                elif "429" in err:
                    # Parse retry_after_seconds from error
                    retry_after = 60 * (attempt + 2)
                    try:
                        if "retry_after_seconds" in err:
                            retry_after = int(err.split("retry_after_seconds\":")[1].split("}")[0])
                    except: pass
                    p(f"  WAIT {user['name']}: {match[4]} -- 429, sleeping {retry_after}s (attempt {attempt+1}/5)")
                    time.sleep(retry_after)
                else:
                    failed += 1
                    p(f"  ERR {user['name']}: {match[4]} -- {err[:100]}")
                    break
            
            time.sleep(8)
    
    p(f"\n=== Done: {done} seeded, {failed} failed ===")

if __name__ == "__main__":
    main()

File: scripts/seed-all.ts
/**
 * Seed fake users with ALL predictions across ALL group stage matches.
 * Run once: npx tsx scripts/seed-all.ts
 */
import "dotenv/config";

const API = process.env.API_URL ?? "http://localhost:8787";

// ── Fake users with distinct personalities ──
const USERS = [
  { name: "F5", bias: "HOME", confRange: [65, 90], takes: true },
  { name: "Aleko", bias: "AWAY", confRange: [30, 60], takes: true },
  { name: "Tita", bias: "DRAW", confRange: [50, 75], takes: false },
  { name: "Rex", bias: "HOME", confRange: [80, 95], takes: true },
  { name: "Nina", bias: "none", confRange: [40, 70], takes: true },
];

// ── Real WC2026 group stage schedule (ESPN) ──
const MATCHES = [
  { date: "2026-06-11", time: "19:00", home: "Mexico", away: "South Africa", matchId: "MEX-SOU" },
  { date: "2026-06-12", time: "02:00", home: "South Korea", away: "Czechia", matchId: "SOU-CZE" },
  { date: "2026-06-12", time: "19:00", home: "Canada", away: "Bosnia-Herzegovina", matchId: "CAN-BOS" },
  { date: "2026-06-13", time: "01:00", home: "United States", away: "Paraguay", matchId: "UNI-PAR" },
  { date: "2026-06-13", time: "19:00", home: "Qatar", away: "Switzerland", matchId: "QAT-SWI" },
  { date: "2026-06-13", time: "22:00", home: "Brazil", away: "Morocco", matchId: "BRA-MOR" },
  { date: "2026-06-14", time: "01:00", home: "Haiti", away: "Scotland", matchId: "HAI-SCO" },
  { date: "2026-06-14", time: "04:00", home: "Australia", away: "Türkiye", matchId: "AUS-TUR" },
  { date: "2026-06-14", time: "17:00", home: "Germany", away: "Curaçao", matchId: "GER-CUR" },
  { date: "2026-06-14", time: "20:00", home: "Netherlands", away: "Japan", matchId: "NET-JAP" },
  { date: "2026-06-14", time: "23:00", home: "Ivory Coast", away: "Ecuador", matchId: "IVO-ECU" },
  { date: "2026-06-15", time: "02:00", home: "Sweden", away: "Tunisia", matchId: "SWE-TUN" },
  { date: "2026-06-15", time: "16:00", home: "Spain", away: "Cape Verde", matchId: "SPA-CAP" },
  { date: "2026-06-15", time: "19:00", home: "Belgium", away: "Egypt", matchId: "BEL-EGY" },
  { date: "2026-06-15", time: "22:00", home: "Saudi Arabia", away: "Uruguay", matchId: "SAU-URU" },
  { date: "2026-06-16", time: "01:00", home: "Iran", away: "New Zealand", matchId: "IRA-NEW" },
  { date: "2026-06-16", time: "19:00", home: "France", away: "Senegal", matchId: "FRA-SEN" },
  { date: "2026-06-16", time: "22:00", home: "Iraq", away: "Norway", matchId: "IRA-NOR" },
  { date: "2026-06-17", time: "01:00", home: "Argentina", away: "Algeria", matchId: "ARG-ALG" },
  { date: "2026-06-17", time: "04:00", home: "Austria", away: "Jordan", matchId: "AUS-JOR" },
  { date: "2026-06-17", time: "17:00", home: "Portugal", away: "Congo DR", matchId: "POR-CON" },
  { date: "2026-06-17", time: "20:00", home: "England", away: "Croatia", matchId: "ENG-CRO" },
  { date: "2026-06-17", time: "23:00", home: "Ghana", away: "Panama", matchId: "GHA-PAN" },
  { date: "2026-06-18", time: "02:00", home: "Uzbekistan", away: "Colombia", matchId: "UZB-COL" },
  { date: "2026-06-18", time: "16:00", home: "Czechia", away: "South Africa", matchId: "CZE-SOU" },
  { date: "2026-06-18", time: "19:00", home: "Switzerland", away: "Bosnia-Herzegovina", matchId: "SWI-BOS" },
  { date: "2026-06-18", time: "22:00", home: "Canada", away: "Qatar", matchId: "CAN-QAT" },
  { date: "2026-06-19", time: "01:00", home: "Mexico", away: "South Korea", matchId: "MEX-SOU1" },
  { date: "2026-06-19", time: "19:00", home: "United States", away: "Australia", matchId: "UNI-AUS" },
  { date: "2026-06-19", time: "22:00", home: "Scotland", away: "Morocco", matchId: "SCO-MOR" },
  { date: "2026-06-20", time: "00:30", home: "Brazil", away: "Haiti", matchId: "BRA-HAI" },
  { date: "2026-06-20", time: "03:00", home: "Türkiye", away: "Paraguay", matchId: "TUR-PAR" },
  { date: "2026-06-20", time: "17:00", home: "Netherlands", away: "Sweden", matchId: "NET-SWE" },
  { date: "2026-06-20", time: "20:00", home: "Germany", away: "Ivory Coast", matchId: "GER-IVO" },
  { date: "2026-06-21", time: "00:00", home: "Ecuador", away: "Curaçao", matchId: "ECU-CUR" },
  { date: "2026-06-21", time: "04:00", home: "Tunisia", away: "Japan", matchId: "TUN-JAP" },
  { date: "2026-06-21", time: "16:00", home: "Spain", away: "Saudi Arabia", matchId: "SPA-SAU" },
  { date: "2026-06-21", time: "19:00", home: "Belgium", away: "Iran", matchId: "BEL-IRA" },
  { date: "2026-06-21", time: "22:00", home: "Uruguay", away: "Cape Verde", matchId: "URU-CAP" },
  { date: "2026-06-22", time: "01:00", home: "New Zealand", away: "Egypt", matchId: "NEW-EGY" },
  { date: "2026-06-22", time: "17:00", home: "Argentina", away: "Austria", matchId: "ARG-AUS" },
  { date: "2026-06-22", time: "21:00", home: "France", away: "Iraq", matchId: "FRA-IRA" },
  { date: "2026-06-23", time: "00:00", home: "Norway", away: "Senegal", matchId: "NOR-SEN" },
  { date: "2026-06-23", time: "03:00", home: "Jordan", away: "Algeria", matchId: "JOR-ALG" },
  { date: "2026-06-23", time: "17:00", home: "Portugal", away: "Uzbekistan", matchId: "POR-UZB" },
  { date: "2026-06-23", time: "20:00", home: "England", away: "Ghana", matchId: "ENG-GHA" },
  { date: "2026-06-23", time: "23:00", home: "Panama", away: "Croatia", matchId: "PAN-CRO" },
  { date: "2026-06-24", time: "02:00", home: "Colombia", away: "Congo DR", matchId: "COL-CON" },
  { date: "2026-06-24", time: "19:00", home: "Bosnia-Herzegovina", away: "Qatar", matchId: "BOS-QAT" },
  { date: "2026-06-24", time: "19:00", home: "Switzerland", away: "Canada", matchId: "SWI-CAN" },
  { date: "2026-06-24", time: "22:00", home: "Morocco", away: "Haiti", matchId: "MOR-HAI" },
  { date: "2026-06-24", time: "22:00", home: "Scotland", away: "Brazil", matchId: "SCO-BRA" },
  { date: "2026-06-25", time: "01:00", home: "Czechia", away: "Mexico", matchId: "CZE-MEX" },
  { date: "2026-06-25", time: "01:00", home: "South Africa", away: "South Korea", matchId: "SOU-SOU" },
  { date: "2026-06-25", time: "20:00", home: "Curaçao", away: "Ivory Coast", matchId: "CUR-IVO" },
  { date: "2026-06-25", time: "20:00", home: "Ecuador", away: "Germany", matchId: "ECU-GER" },
  { date: "2026-06-25", time: "23:00", home: "Japan", away: "Sweden", matchId: "JAP-SWE" },
  { date: "2026-06-25", time: "23:00", home: "Tunisia", away: "Netherlands", matchId: "TUN-NET" },
  { date: "2026-06-26", time: "02:00", home: "Paraguay", away: "Australia", matchId: "PAR-AUS" },
  { date: "2026-06-26", time: "02:00", home: "Türkiye", away: "United States", matchId: "TUR-UNI" },
  { date: "2026-06-26", time: "19:00", home: "Norway", away: "France", matchId: "NOR-FRA" },
  { date: "2026-06-26", time: "19:00", home: "Senegal", away: "Iraq", matchId: "SEN-IRA" },
  { date: "2026-06-27", time: "00:00", home: "Cape Verde", away: "Saudi Arabia", matchId: "CAP-SAU" },
  { date: "2026-06-27", time: "00:00", home: "Uruguay", away: "Spain", matchId: "URU-SPA" },
  { date: "2026-06-27", time: "03:00", home: "Egypt", away: "Iran", matchId: "EGY-IRA" },
  { date: "2026-06-27", time: "03:00", home: "New Zealand", away: "Belgium", matchId: "NEW-BEL" },
  { date: "2026-06-27", time: "21:00", home: "Croatia", away: "Ghana", matchId: "CRO-GHA" },
  { date: "2026-06-27", time: "21:00", home: "Panama", away: "England", matchId: "PAN-ENG" },
  { date: "2026-06-27", time: "23:30", home: "Colombia", away: "Portugal", matchId: "COL-POR" },
  { date: "2026-06-27", time: "23:30", home: "Congo DR", away: "Uzbekistan", matchId: "CON-UZB" },
  { date: "2026-06-28", time: "02:00", home: "Algeria", away: "Austria", matchId: "ALG-AUS" },
  { date: "2026-06-28", time: "02:00", home: "Jordan", away: "Argentina", matchId: "JOR-ARG" },
];

// Hot takes per user
const TAKES: Record<string, string[]> = {
  F5: [
    "Mexico opening night at Banorte, crowd goes insane",
    "South Korea always shows up in WC",
    "Canada hosting is a game changer",
    "USA at SoFi, home advantage is real",
    "Brazil vs Morocco is the match of the group",
    "Germany will destroy Curaçao",
    "Netherlands vs Japan is quality",
    "Spain too strong for Cape Verde",
    "Belgium's last dance with this generation",
    "France cruises through group",
    "England vs Croatia 2018 rematch",
    "Argentina still dangerous even post-Messi era",
    "Portugal destroys Congo DR",
    "Turkey at home soil, dangerous",
    "Uruguay always peaks in World Cups",
    "Japan beats Tunisia easily",
    "Brazil only needs 3 matches to top the group",
    "Colombia dark horse of the tournament",
    "USA beats Australia, book it",
    "Norway with Haaland is scary",
    "England vs Ghana will be interesting",
    "Croatia still has Modric magic",
    "Switzerland always makes knockout rounds",
    "Morocco rides 2022 momentum",
  ],
  Aleko: [
    "Czechia pulls off the upset vs South Korea",
    "Bosnia shocks Canada at BMO Field",
    "Paraguay takes points off USA",
    "Qatar proves 2022 wasn't a fluke",
    "Haiti beats Scotland, Caribbean pride",
    "Türkiye upsets Australia on home turf",
    "Ivory Coast silences Ecuador",
    "Tunisia gets result vs Sweden",
    "Saudi Arabia shocks Uruguay again",
    "New Zealand draws Iran, Kiwi resilience",
    "Iraq surprises Norway",
    "Jordan takes a point off Algeria",
    "Congo DR upsets Portugal",
    "Ghana beats Panama in opener",
    "Uzbekistan stuns Colombia",
    "Czechia beats South Africa comfortably",
    "Haiti draws Scotland again",
    "Ecuador beats Germany, calling it",
    "Japan beats Netherlands",
    "Tunisia takes a point off Netherlands",
    "Panama draws Croatia",
    "Congo DR beats Uzbekistan",
    "Jordan upsets Argentina",
    "Algeria takes down Austria",
  ],
  Tita: [
    "Draw. Definitely a draw.",
    "Low scoring, 1-1 at most",
    "Neither team breaks through",
    "Defensive affair, shared points",
    "Safe pick, both teams evenly matched",
    "0-0 written all over this one",
    "Boring game but a point is valuable",
    "Both teams playing not to lose",
    "This screams draw to me",
    "Set piece decides it, but it ends level",
    "Neither side has the cutting edge",
    "Conservative tactics, draw incoming",
  ],
  Rex: [
    "Mexico 2-0, easy opener",
    "South Korea 1-0, clinical finish",
    "Canada 3-1, home crowd carries them",
    "USA dominates Paraguay, 3-0",
    "Qatar is free points for Switzerland",
    "Brazil 4-0 Morocco, not even close",
    "Haiti can't handle Scotland",
    "Germany obliterates Curaçao, 5-0",
    "Netherlands 2-0 Japan, total football",
    "Spain 4-0 Cape Verde, stroll in the park",
    "Belgium 2-0 Egypt, golden gen farewell tour",
    "Uruguay 3-0 Saudi Arabia",
    "France 3-0 Senegal, Mbappe feast",
    "Argentina 2-0 Algeria, statement win",
    "Portugal 3-0 Congo DR, cruise control",
    "England 2-0 Croatia, revenge game",
    "Brazil cruises past Haiti 4-0",
    "Germany 3-0 Ivory Coast",
    "Argentina 2-0 Austria",
    "France 3-1 Iraq",
    "Portugal demolishes Uzbekistan 4-0",
    "England beats Ghana 3-0",
    "Norway 2-1 Senegal, Haaland brace",
    "Colombia 2-0 Congo DR",
  ],
  Nina: [],
};

function pick<T>(arr: T[]): T {
  return arr[Math.floor(Math.random() * arr.length)];
}

function randInt(min: number, max: number): number {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

function pickWinner(user: typeof USERS[0]): "HOME" | "AWAY" | "DRAW" {
  if (user.bias === "HOME") return "HOME";
  if (user.bias === "AWAY") return "AWAY";
  if (user.bias === "DRAW") return "DRAW";
  return pick(["HOME", "AWAY", "DRAW"] as const);
}

async function predict(user: string, match: typeof MATCHES[0], win: string, confidence: number, take: string, attempt = 1): Promise<boolean> {
  try {
    const r = await fetch(`${API}/api/predict`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ user, matchId: match.matchId, home: match.home, away: match.away, pick: win, confidence, take }),
    });
    const data = await r.json();
    if (data.error) {
      if ((data.error.includes("429") || data.error.includes("Rate limit") || data.error.includes("retry_after")) && attempt < 4) {
        const wait = 65 * attempt;
        console.log(`  ⏳ ${user}: ${match.matchId} — rate limited, waiting ${wait}s (attempt ${attempt}/3)...`);
        await new Promise(r => setTimeout(r, wait * 1000));
        return predict(user, match, win, confidence, take, attempt + 1);
      }
      console.error(`  ✗ ${user}: ${match.matchId} — ${data.error}`);
      return false;
    }
    console.log(`  ✓ ${user}: ${match.matchId} → ${win} (${confidence}%)${take ? ' "' + take.substring(0, 40) + '..."' : ""}`);
    return true;
  } catch (e: any) {
    console.error(`  ✗ ${user}: ${match.matchId} — ${e.message}`);
    return false;
  }
}

async function main() {
  console.log(`\n=== Seeding ALL ${MATCHES.length} matches × ${USERS.length} users = ${MATCHES.length * USERS.length} predictions ===\n`);
  console.log(`Delay: 2.5s between requests (30 req/min limit)`);
  console.log(`Estimated time: ${Math.ceil(MATCHES.length * USERS.length * 2.5 / 60)} minutes\n`);

  let total = 0;
  let failed = 0;

  for (const user of USERS) {
    console.log(`\n── ${user.name} (${user.bias} bias) ──`);
    const takes = user.takes && TAKES[user.name] ? [...TAKES[user.name]] : [];

    for (const match of MATCHES) {
      const win = pickWinner(user);
      const conf = randInt(user.confRange[0], user.confRange[1]);
      const take = takes.length > 0 ? pick(takes) : "";
      if (take) {
        const idx = takes.indexOf(take);
        if (idx > -1) takes.splice(idx, 1);
      }

      const ok = await predict(user.name, match, win, conf, take);
      if (ok) total++;
      else failed++;

      await new Promise(r => setTimeout(r, 2500));
    }
  }

  console.log(`\n=== Done: ${total} predictions seeded, ${failed} failed ===`);
}

main().catch(console.error);
